pyanaconda/ui/gui/spokes/installation_source.py

```diff
@@ -922,7 +955,18 @@ def refresh(self):
         self._protocol_combo_box.set_active_id(PROTOCOL_MIRROR)
         self._url_type_combo_box.set_active_id(URL_TYPE_BASEURL)
 
-        if source_type == SOURCE_TYPE_URL:
+        # unset CDN as installation source if the system has been
+        # unregistered in the meantime
+        cdn_button_active = self._cdn_button.get_active()
+        if cdn_button_active and not source_type == SOURCE_TYPE_CDN:
```

Comment:
This logic is chaotic. Why you cannot set up the source and let the payload restart fail?

Reply:
I might be getting a bit lost here - how is this related to the payload restart ?

These conditionals make sure the CDN radio button is enabled/disabled based on CDN being set as installation source and the other conditionals do the same. The one specifics o the CDN source is that in some cases (CDROM source) the CDN source might be switched to another source (CDROM) outside of the Source spoke and we need to handle that.

Reply:
My mistake, it is not related to the payload restart. However, this logic is still too complicated for no reason.

| Is the button active? | Is the current type CDN? | Should the button stay active? | 
| --- | --- | --- |
| 0 | 0 | 0 |
| 0 | 1 | 1 |
| 1 | 0 | 0 |
| 1 | 1 | 1 |

Based on the table, the button activation depends only on the current type of the source. Therefore, you can replace this code:
```
        cdn_button_active = self._cdn_button.get_active()
        if cdn_button_active and not source_type == SOURCE_TYPE_CDN:
            log.debug("source spoke: unregistered since last visit, turning off CDN button")
            self._cdn_button.set_active(False)

        # Set CDN radio button active, in case external conditions changed since the last refresh.
        if source_type == SOURCE_TYPE_CDN and not self._cdn_button.get_active():
            log.debug("source spoke: CDN enabled, enabling CDN button")
            self._cdn_button.set_active(True)
        elif source_type == SOURCE_TYPE_URL:
            ...
```

With this one:

```    
        if source_type == SOURCE_TYPE_CDN:
            self._cdn_button.set_active(True)
        elif source_type == SOURCE_TYPE_URL:
            ...

```

Reply:
I don't think this is correct - how do you turn the button OFF when CDN is not the installation source at spoke entry with this ? This what the first condition does.

Ideally, there would be just:

`self._cdn_button.set_active(source_type == SOURCE_TYPE_URL)`

Still, then the if/else tree down there would get executed resulting in:

```
        else:
            ValueError("Unsupported source type: '{}'".format(source_type))
```

Reply:
Actually, a lot of this is no longer needed with the rebased Anaconda - in 8.2 there wasn't really a CDN source and the CDN radio button held part of the state.

Now there is a proper CDN source, so all of these previous considerations are not really needed anymore and a simple condition should be enough. :)